runelite-client/src/main/java/net/runelite/client/plugins/barrows/BarrowsPlugin.java

```diff
@@ -243,5 +249,29 @@ public void onWidgetLoaded(WidgetLoaded event)
 				.runeLiteFormattedMessage(message.build())
 				.build());
 		}
+
+		if (event.getGroupId() == WidgetID.BARROWS_PUZZLE_GROUP_ID)
+		{
+			final int answer = client.getWidget(WidgetInfo.BARROWS_FIRST_PUZZLE).getModelId() - 3;
+			final ImmutableSet<WidgetInfo> possibleSolutions = ImmutableSet.of(
+					WidgetInfo.BARROWS_PUZZLE_ANSWER1,
+					WidgetInfo.BARROWS_PUZZLE_ANSWER2,
+					WidgetInfo.BARROWS_PUZZLE_ANSWER3
+			);
+
+			for (WidgetInfo puzzleNode : possibleSolutions)
+			{
+				if (puzzleNode == null)
+				{
+					continue;
+				}
+
+				if (client.getWidget(puzzleNode).getModelId() == answer)
+				{
+					widgetToHighlight = client.getWidget(puzzleNode);
```

Comment:
also what if no widget is found? you should set widgetToHighlight to null before looping probably or figure out a way to reset it properly